Testing API throttling is inconsistent

### Context

When we test throttling API requests it doesn't actually throttle every time we run our tests.

The code in question in `test_api_general.py`
```python
for i in range(15):
   response = self.client.get(reverse("survey_jobs", kwargs={"version": API_VERSION}))
self.assertEqual(response.status_code, status.HTTP_429_TOO_MANY_REQUESTS)
```


### Problem or idea

Output from a non-throttled test failing:

```console
System check identified no issues (0 silenced).
..........F.......................Deleting index 'experiments'
Creating index 'experiments'
Indexing 2 'Experiment' objects 
.....
======================================================================
FAIL: test_all_endpoints (tests.views.test_api_general.APITestCases)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/user/tests/views/test_api_general.py", line 352, in test_all_endpoints
    self.assertEqual(response.status_code, status.HTTP_429_TOO_MANY_REQUESTS)
AssertionError: 200 != 429

----------------------------------------------------------------------
Ran 39 tests in 8.177s
```

### Solution or next step

We should rewrite the test so that it triggers a throttled response at the end or collect the responses and check to see if throttling occurred.

--- a/api/tests/views/test_api_general.py
+++ b/api/tests/views/test_api_general.py
@@ -347,9 +347,14 @@ class APITestCases(APITestCase):
 
         # Tenth call since reset_cache() should be throttled, three have happened.
         # Make more than necessary to ensure we get the throttle.
-        for i in range(15):
+        response_codes = []
+        for _ in range(15):
             response = self.client.get(reverse("survey_jobs", kwargs={"version": API_VERSION}))
-        self.assertEqual(response.status_code, status.HTTP_429_TOO_MANY_REQUESTS)
+            response_codes.append(response.status_code)
+        self.assertGreater(
+            len([rc for rc in response_codes if rc == status.HTTP_429_TOO_MANY_REQUESTS]),
+            0,
+        )
 
     def test_experiment_multiple_accessions(self):
         response = self.client.get(

